src/main/java/org/broadinstitute/consent/http/resources/UserResource.java

```diff
@@ -53,6 +57,38 @@ public UserResource(UserService userService, LibraryCardService libraryCardServi
         this.gson = new Gson();
     }
 
+    @GET
+    @Produces("application/json")
+    @Path("/role/{roleName}")
+    @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
+    public Response getUsers(@Auth AuthUser authUser, @PathParam("roleName") String roleName) {
```

Comment:
Instead of having UserRole validations/ thrown Exceptions happen on the resource it may be better to have it contained within service methods. It'll keep the business logic contained in reusable functions while making the Resource logic smaller and easier to follow. 

On a side note the other methods on UserResource are good candidates for refactoring for a later tech debt ticket.

Reply:
I just had a very similar PR to this but for dars instead of users, DUOS-1259, where I had originally done this same logic and validation in the service class bc that was my thought too, but on the suggestion of Greg I moved it to the resource class, so I was continuing to follow that feedback here. It seems the same to me, so thoughts/preferences? @JVThomas @rushtong

Reply:
@JVThomas It was my feedback to put the role validation logic here in the resource. The resource level is the right place to block entry to invalid requests which is why we have the role permissions here and in other resource classes, we validate that the user is associated to an entity like a DAR. The service class shouldn't have to deal with that level of detail and focus on doing the processing of the entities.

Reply:
The logic seems extractable and reusable for multiple endpoints. You can have the logic contained and testable in a separate function and still use it within the resource so that it (resource) can maintain its responsibility of proper authorization.

It'll also makes testing easier - the encapsulated function can be tested for its business logic (role validation) while the resource can be tested to ensure it returns proper responses.